### PR TITLE
Added new dimension to internal metrics to identify mode

### PR DESCRIPTION
**Pull request by rcastley**
**Description:**

To easily filter between `agent` and `gateway` deployments of the collector in dashboards a proposed new dimension `otelcol.service.mode` to be added to the `metrics/internal` pipeline.

**Link to Splunk idea:**

N/A

**Testing:**

Deployed the collector with the configuration changes and successfully filtered collector mode in OOTB OTel Dashboard in O11y Cloud.

**Documentation:**

N/A

### CODE REVIEW

**Comment by hhgsplk** [comment]
This is a great way for customers that want to check the load on all their otel gateways. Recently came up during a customer project. Please approve.

**Comment by atoulme** [comment]
CI fails. We will review.

File: cmd/otelcol/config/collector/agent_config.yaml [hunk 1]
```diff
@@ -123,7 +123,11 @@ processors:
       #- action: insert
         #value: staging/production/...
         #key: deployment.environment
-
+  resource/add_mode:
+    attributes:
+      - action: insert
+        value: "agent"
+        key: service_mode
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/agent_config.yaml:130]: I know there is no existing conventions for this, but OTel spec recommends the resource attribute names to be specific enough to figure out what entity they are part of, using namespacing. I think something like `otelcol.service.mode` whould be better choice here.

Comment (rcastley) [reply cmd/otelcol/config/collector/agent_config.yaml]: PR updated to relfect the semantic naming conventions, thank you for your insight ... I originally went off `service_version`.

File: cmd/otelcol/config/collector/agent_config.yaml [hunk 2]
```diff
@@ -123,7 +123,11 @@ processors:
       #- action: insert
         #value: staging/production/...
         #key: deployment.environment
-
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/agent_config.yaml:126]: nit: let's keep the space before the `exporters` section

**Review by dmitryax — APPROVED** [review]

File: cmd/otelcol/config/collector/agent_config.yaml (hunk 2 above)
Comment (rcastley) [reply cmd/otelcol/config/collector/agent_config.yaml:126]: Changes committed

**Review by pjanotti — APPROVED** [review]

File: cmd/otelcol/config/collector/agent_config.yaml (hunk 2 above)
Comment (rcastley) [reply cmd/otelcol/config/collector/agent_config.yaml:126]: @hughesjj Yes, only needed on the OOTB OTel dashboards

**Comment by rcastley** [comment]
> @rcastley the `TestDefaultGatewayConfig` are broken because they need to be updated whenever the config yamls are changed. Let's know if you need help with that.
> 
> https://github.com/signalfx/splunk-otel-collector/actions/runs/12303965814/job/34340867187#step:16:141:~:text=default_config_test.go%3A59,Test%3A%20%20%20%20%20%20%20%09TestDefaultGatewayConfig/default

@pjanotti I have tried and failed 👎🏻 I updated the `default_config_test.go` for both agent and gateway modes but it is still failing on two tests. If you could help me to understand why that would be awesome. Thank you :-)

File: tests/general/default_config_test.go [hunk 3]
```diff
@@ -118,6 +118,13 @@ func TestDefaultGatewayConfig(t *testing.T) {
 						"detectors": []any{"gcp", "ecs", "ec2", "azure", "system"},
 						"override":  true,
 					},
+					"resource/add_mode": map[string]any{
+						"attributes": map[string]any{
+							"action": "insert",
+							"value":  "gateway",
+							"key":    "otelcol.service.node",
```
Comment (jinja2) [inline tests/general/default_config_test.go:125]:
```suggestion
							"key":    "otelcol.service.mode",
```

File: tests/general/default_config_test.go [hunk 4]
```diff
@@ -314,6 +321,13 @@ func TestDefaultAgentConfig(t *testing.T) {
 						"detectors": []any{"gcp", "ecs", "ec2", "azure", "system"},
 						"override":  true,
 					},
+					"resource/add_mode": map[string]any{
+						"attributes": map[string]any{
+							"action": "insert",
+							"value":  "agent",
+							"key":    "otelcol.service.node",
```
Comment (jinja2) [inline tests/general/default_config_test.go:328]:
```suggestion
							"key":    "otelcol.service.mode",
```

File: tests/general/default_config_test.go (hunk 3 above)
Comment (rcastley) [reply tests/general/default_config_test.go:125]: @jinja2 Thank you :-)

**Comment by rcastley** [comment]
Both `Integration Test / integration-test-docker (amd64, integration) (pull_request)` and `Integration Test / integration-test-docker (arm64, integration) (pull_request)` fail with the following:

```
PASS
ok  	github.com/signalfx/splunk-otel-collector/tests/receivers/scriptedinputs	34.078s
?   	github.com/signalfx/splunk-otel-collector/tests/testutils	[no test files]
?   	github.com/signalfx/splunk-otel-collector/tests/testutils/subprocess	[no test files]
?   	github.com/signalfx/splunk-otel-collector/tests/testutils/telemetry	[no test files]
FAIL
make: *** [Makefile:88: integration-test] Error 1
```

I don't understand/know how to resolve this so if someone could help me, I would be most grateful. Thank you!

File: tests/general/default_config_test.go (hunk 3 above)
Comment (jinja2) [reply tests/general/default_config_test.go:125]: Missed committing this

File: tests/general/default_config_test.go [hunk 5]
```diff
@@ -118,6 +118,13 @@ func TestDefaultGatewayConfig(t *testing.T) {
 						"detectors": []any{"gcp", "ecs", "ec2", "azure", "system"},
 						"override":  true,
 					},
+					"resource/add_mode": map[string]any{
+						"attributes": map[string]any{
+							"action": "insert",
+							"value":  "gateway",
+							"key":    "otelcol.service.mode",
+						},
```
Comment (jinja2) [inline tests/general/default_config_test.go:126]:
```suggestion
					"resource/add_mode": map[string]any{
						"attributes": []any{
							map[string]any{
								"action": "insert",
								"value":  "gateway",
								"key":    "otelcol.service.mode",
							},
						},
```

**Comment by rcastley** [comment]
@jinja2 Does the following change:

```
"attributes": []any{
	map[string]any{
		"action": "insert",
		"value":  "gateway",
		"key":    "otelcol.service.mode",
```

need to be applied also to `TestDefaultAgentConfig` function?

File: tests/general/default_config_test.go [hunk 6]
```diff
@@ -314,6 +323,13 @@ func TestDefaultAgentConfig(t *testing.T) {
 						"detectors": []any{"gcp", "ecs", "ec2", "azure", "system"},
 						"override":  true,
 					},
+					"resource/add_mode": map[string]any{
+						"attributes": map[string]any{
+							"action": "insert",
+							"value":  "agent",
+							"key":    "otelcol.service.mode",
+						},
+					},
```
Comment (jinja2) [inline tests/general/default_config_test.go:332]:
```suggestion
					"resource/add_mode": map[string]any{
						"attributes": []any{
							map[string]any{
								"action": "insert",
								"value":  "agent",
								"key":    "otelcol.service.mode",
							},
						},
					},
```

**Comment by jinja2** [comment]
Those failing tests passed 🎉

**Review by jinja2 — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Comment by jinja2** [comment]
@rcastley I think you'll need to setup commit signing https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification. I see `The base branch requires all commits to be signed.` under merging blocked.

**Pull request closed without merging**